android/java/org/chromium/chrome/browser/rate/RateUtils.java

```diff
@@ -25,25 +28,38 @@ public class RateUtils {
 
     public static final String FROM_SETTINGS = "from_settings";
 
-    private static final int DAYS_60 = 60;
-    private static final int APP_OPEN_14 = 14;
-    private static final int APP_OPEN_41 = 41;
-    private static final int APP_OPEN_121 = 121;
+    private static final int DAYS_30 = 30;
+    private static final int APP_OPEN_5 = 5;
+    private static final int BOOKMARKS_COUNT = 5;
+    private static final int LAST_7_DAYS = 7;
 
     private static final String PREF_RATE = "rate";
     private static final String PREF_NEXT_RATE_DATE = "next_rate_date";
     private static final String PREF_NEXT_APP_OPEN_COUNT = "next_app_open_count";
+    private static final String PREF_ADDED_BOOKMARK_COUNT = "added_bookmark_count";
+
+    private static final String PREF_LAST_SESSION_SHOWN = "last_session_shown";
+
+    private static final String PREF_LAST_TIME_APP_USED_DATE1 = "last_time_app_used_date1";
```

Comment:
instead of using 4 different pref, can we use last open timestamp and compare with the current timestamp and determine if it's consecutive days ?

Reply:
Here current timeStamp need to check last 3 timeStamp, if all are less than 7 means , 4 days used in last seven days.
For that need to keep 4 different timestamp in pref.